File: ios/TestTabApp/ShowPostEvent.m
//
//  ShowPostEvent.m
//  TestTabApp
//
//  Created by Branding Brand on 2/20/23.
//

#import "ShowPostEvent.h"

@implementation ShowPostEvent

- (instancetype)initWithEventData:(NSDictionary *)data
{
    self = [super init];
    if (self) {
        NSString *component = [data objectForKey:@"component"];
        NSDictionary *props = [data objectForKey:@"data"];
        _props = [[ShowPostProps alloc] initWithComponentName:component data:props];
    }
    return self;
}

- (BOOL)isValidEvent
{
    if (self.props != nil && [self.props areValidProps]) {
        return true;
    }
    else {
        return false;
    }
}

@end

File: android/app/src/main/java/com/brandingbrand/testtabapp/reactgateway/ShowPostEvent.kt
package com.brandingbrand.testtabapp.reactgateway

import com.facebook.react.bridge.ReadableMap

class ShowPostEvent(data: ReadableMap) {
    val props: ShowPostProps = ShowPostProps(data)
    val invokingComponent: String? = data.getString("component")

    fun isValidEvent(): Boolean {
        if (props.areValidProps() &&
            !invokingComponent.isNullOrEmpty()) {
            return true
        }
        return false
    }
}